core/app/models/spree/stock_item.rb

```diff
@@ -6,12 +6,15 @@ class StockItem < Spree::Base
     belongs_to :variant, class_name: 'Spree::Variant', inverse_of: :stock_items
     has_many :stock_movements, inverse_of: :stock_item
 
-    validates_presence_of :stock_location, :variant
-    validates_uniqueness_of :variant_id, scope: [:stock_location_id, :deleted_at]
+    validates :stock_location, :variant, presence: true
+    validates :variant_id, uniqueness: { scope: [:stock_location_id, :deleted_at] }, allow_blank: true
```

Comment:
Did we want this `allow_blank: true` to be a part of the uniqueness clause?
`validates :variant_id, uniqueness: { scope: [:stock_location_id, :deleted_at], allow_blank: true }`